Makefile

```diff
@@ -1,98 +1,151 @@
+
+install: ## Install the OpenFGA service
+	${call print, "Installing the OpenFGA binary within ${GO_BIN}"}
+	@go install -v "$(CURDIR)/cmd/${BINARY_NAME}"
+
+#-----------------------------------------------------------------------------------------------------------------------
+# Checks
+#-----------------------------------------------------------------------------------------------------------------------
 .PHONY: lint
+
 lint: $(GO_BIN)/golangci-lint ## Lint Go source files
-	@echo "==> Linting Go source files"
+	${call print, "Linting Go source files"}
 	@golangci-lint run -v --fix -c .golangci.yaml ./...
 
-.PHONY: clean
-clean: ## Clean files
-	rm ./openfga
+#-----------------------------------------------------------------------------------------------------------------------
+# Tests
+#-----------------------------------------------------------------------------------------------------------------------
+.PHONY: test test-docker test-bench test-mocks
 
-.PHONY: build
-build: ## Build the OpenFGA service
-	go build -o ./openfga ./cmd/openfga
-
-.PHONY: run
-run: build ## Run the OpenFGA server with in-memory storage
-	./openfga run
-
-.PHONY: start-postgres
-start-postgres: ## Start a Postgres Docker container
-	docker run -d --name postgres -p 5432:5432 -e POSTGRES_USER=postgres -e POSTGRES_PASSWORD=password postgres:14
-	@echo \> use \'postgres://postgres:password@localhost:5432/postgres\' to connect to postgres
-
-.PHONY: migrate-postgres
-migrate-postgres: build ## Run Postgres migrations
-	# nosemgrep: detected-username-and-password-in-uri
-	./openfga migrate --datastore-engine postgres --datastore-uri 'postgres://postgres:password@localhost:5432/postgres'
-
-.PHONY: run-postgres
-run-postgres: build ## Run the OpenFGA server with Postgres storage
-	./openfga run --datastore-engine postgres --datastore-uri 'postgres://postgres:password@localhost:5432/postgres'
-
-.PHONY: start-mysql
-start-mysql: build ## Start a MySQL Docker container
-	docker run -d --name mysql -p 3306:3306 -e MYSQL_ROOT_PASSWORD=secret -e MYSQL_DATABASE=openfga mysql:8
-
-.PHONY: migrate-mysql
-migrate-mysql: build ## Run MySQL migrations
-	# nosemgrep: detected-username-and-password-in-uri
-	./openfga migrate --datastore-engine mysql --datastore-uri 'root:secret@tcp(localhost:3306)/openfga?parseTime=true'
-
-.PHONY: run-mysql
-run-mysql: build ## Run the OpenFGA server with MySQL storage
-	./openfga run --datastore-engine mysql --datastore-uri 'root:secret@tcp(localhost:3306)/openfga?parseTime=true'
-
-.PHONY: download
-download:
-	@cd tools/ && go mod download
-
-.PHONY: install-tools
-install-tools: download ## Install developer tooling
-	@cd tools && go list -e -f '{{range .Imports}}{{.}} {{end}}' tools.go | CGO_ENABLED=0 xargs go install -mod=readonly
-
-.PHONY: go-generate
-go-generate: install-tools
-	go generate ./...
-
-.PHONY: test
-test: go-generate ## Run all tests
-	go test -race \
-			-coverpkg=./... \
+test: test-mocks ## Run all tests
```

Comment:
`make test` got too verbose

```
ok      github.com/openfga/openfga/tests/listobjects    23.958s coverage: 52.4% of statements in github.com/openfga/openfga/assets, github.com/openfga/openfga/cmd, github.com/openfga/openfga/cmd/migrate, github.com/openfga/openfga/cmd/openfga, github.com/openfga/openfga/cmd/run, github.com/openfga/openfga/cmd/util, github.com/openfga/openfga/cmd/validatemodels, github.com/openfga/openfga/internal/authn, github.com/openfga/openfga/internal/authn/oidc, github.com/openfga/openfga/internal/authn/presharedkey, github.com/openfga/openfga/internal/build, github.com/openfga/openfga/internal/condition, github.com/openfga/openfga/internal/condition/eval, github.com/openfga/openfga/internal/condition/metrics, github.com/openfga/openfga/internal/condition/types, github.com/openfga/openfga/internal/graph, github.com/openfga/openfga/internal/keys, github.com/openfga/openfga/internal/middleware/authn, github.com/openfga/openfga/internal/mocks, github.com/openfga/openfga/internal/server/config, github.com/openfga/openfga/internal/test/check, github.com/openfga/openfga/internal/test/listobjects, github.com/openfga/openfga/internal/utils, github.com/openfga/openfga/internal/validation, github.com/openfga/openfga/pkg/encoder, github.com/openfga/openfga/pkg/encrypter, github.com/openfga/openfga/pkg/gateway, github.com/openfga/openfga/pkg/logger, github.com/openfga/openfga/pkg/middleware/http, github.com/openfga/openfga/pkg/middleware/logging, github.com/openfga/openfga/pkg/middleware/requestid, github.com/openfga/openfga/pkg/middleware/storeid, github.com/openfga/openfga/pkg/middleware/validator, github.com/openfga/openfga/pkg/server, github.com/openfga/openfga/pkg/server/commands, github.com/openfga/openfga/pkg/server/commands/reverseexpand, github.com/openfga/openfga/pkg/server/errors, github.com/openfga/openfga/pkg/server/health, github.com/openfga/openfga/pkg/server/test, github.com/openfga/openfga/pkg/storage, github.com/openfga/openfga/pkg/storage/memory, github.com/openfga/openfga/pkg/storage/mysql, github.com/openfga/openfga/pkg/storage/postgres, github.com/openfga/openfga/pkg/storage/sqlcommon, github.com/openfga/openfga/pkg/storage/storagewrappers, github.com/openfga/openfga/pkg/storage/test, github.com/openfga/openfga/pkg/telemetry, github.com/openfga/openfga/pkg/testfixtures/storage, github.com/openfga/openfga/pkg/testutils, github.com/openfga/openfga/pkg/tuple, github.com/openfga/openfga/pkg/typesystem, github.com/openfga/openfga/tests, github.com/openfga/openfga/tests/check, github.com/openfga/openfga/tests/listobjects, github.com/openfga/openfga/tests/writemodel
ok      github.com/openfga/openfga/tests/writemodel     4.668s  coverage: 15.1% of statements in github.com/openfga/openfga/assets, github.com/openfga/openfga/cmd, github.com/openfga/openfga/cmd/migrate, github.com/openfga/openfga/cmd/openfga, github.com/openfga/openfga/cmd/run, github.com/openfga/openfga/cmd/util, github.com/openfga/openfga/cmd/validatemodels, github.com/openfga/openfga/internal/authn, github.com/openfga/openfga/internal/authn/oidc, github.com/openfga/openfga/internal/authn/presharedkey, github.com/openfga/openfga/internal/build, github.com/openfga/openfga/internal/condition, github.com/openfga/openfga/internal/condition/eval, github.com/openfga/openfga/internal/condition/metrics, github.com/openfga/openfga/internal/condition/types, github.com/openfga/openfga/internal/graph, github.com/openfga/openfga/internal/keys, github.com/openfga/openfga/internal/middleware/authn, github.com/openfga/openfga/internal/mocks, github.com/openfga/openfga/internal/server/config, github.com/openfga/openfga/internal/test/check, github.com/openfga/openfga/internal/test/listobjects, github.com/openfga/openfga/internal/utils, github.com/openfga/openfga/internal/validation, github.com/openfga/openfga/pkg/encoder, github.com/openfga/openfga/pkg/encrypter, github.com/openfga/openfga/pkg/gateway, github.com/openfga/openfga/pkg/logger, github.com/openfga/openfga/pkg/middleware/http, github.com/openfga/openfga/pkg/middleware/logging, github.com/openfga/openfga/pkg/middleware/requestid, github.com/openfga/openfga/pkg/middleware/storeid, github.com/openfga/openfga/pkg/middleware/validator, github.com/openfga/openfga/pkg/server, github.com/openfga/openfga/pkg/server/commands, github.com/openfga/openfga/pkg/server/commands/reverseexpand, github.com/openfga/openfga/pkg/server/errors, github.com/openfga/openfga/pkg/server/health, github.com/openfga/openfga/pkg/server/test, github.com/openfga/openfga/pkg/storage, github.com/openfga/openfga/pkg/storage/memory, github.com/openfga/openfga/pkg/storage/mysql, github.com/openfga/openfga/pkg/storage/postgres, github.com/openfga/openfga/pkg/storage/sqlcommon, github.com/openfga/openfga/pkg/storage/storagewrappers, github.com/openfga/openfga/pkg/storage/test, github.com/openfga/openfga/pkg/telemetry, github.com/openfga/openfga/pkg/testfixtures/storage, github.com/openfga/openfga/pkg/testutils, github.com/openfga/openfga/pkg/tuple, github.com/openfga/openfga/pkg/typesystem, github.com/openfga/openfga/tests, github.com/openfga/openfga/tests/check, github.com/openfga/openfga/tests/listobjects, github.com/openfga/openfga/tests/writemodel
[15/03/24 12:00:31] ~/GitHub/openfga (dev-flow-improvements) 
```

Reply:
It's a bit unclear if your comment is just a note / suggestion / blocker, could you help clarify 🙏🏻 ? 

That's due to the fact that the coverpkg was changed to equal the packages that are actually testable and coverable instead of using `./...`. If you prefer I can switch it back to that so you don't see a list of all. If we do it as it is right now we don't need to grep -v mocks any more but the output is a bit more verbose.

Reply:
I can also add a filter capability to the make command to also allow just running 1 test at a time using a regex.

Reply:
@miparnisari addressed in [`fb1456e` (#1454)](https://github.com/openfga/openfga/pull/1454/commits/fb1456e8c3eb9759f8255700760eb87b8f17a90a)

Reply:
> It's a bit unclear if your comment is just a note / suggestion / blocker, could you help clarify 🙏🏻 ?

Ahhh sorry about that. It's not a blocker, just something that i really don't like 😆 i like the current compact output.

Could we also get rid of

<img width="616" alt="image" src="https://github.com/openfga/openfga/assets/5374887/cbecf1f8-0b05-4b34-afe5-5254e002889f">

?

Reply:
[`36d14d2` (#1454)](https://github.com/openfga/openfga/pull/1454/commits/36d14d25a50c94cf7e192fb6d7529093aeacc90a)